src/main/java/serviceimpl/TransactionParserImpl.java

```diff
@@ -0,0 +1,39 @@
+package serviceimpl;
+
+import db.Storage;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.TransactionParserService;
+import strategy.StrategyChoosing;
+
+public class TransactionParserImpl implements TransactionParserService {
+    private static final int FIRST_VALID_LINE = 1;
+    private static final int COLUMN_QUANTITY_CHECKER = 3;
+    private static final String DATA_SEPARATOR = ",";
+    private static final int OPERATION_CODE = 0;
+    private static final int FRUIT_NAME = 1;
+    private static final int FRUIT_QUANTITY = 2;
+    private final StrategyChoosing strategyChoosing;
+
+    public TransactionParserImpl(StrategyChoosing strategyChoosing) {
+        this.strategyChoosing = strategyChoosing;
+    }

```

Comment:
You do not need a strategy here, it should be either a separate `TransactionEvaluator` or a loop iterating over the transactions list in the main method
`TransactionParser` is just a mapper to turn a list of strings into a list of transactions without any storage or strategy